src/software/thunderscope/wifi_communication_manager.py

```diff
@@ -43,12 +50,12 @@ def __init__(
         """
         ## Robot IP address tracking ##
         self.robot_ip_addresses: list[tuple[Lock, str | None]] = [
-            (Lock(), None) for _ in range(MAX_ROBOT_IDS_PER_SIDE)
+            (Lock(), None) for ip in robot_ip
```

Comment:
I would suggest structuring it this way:
```suggestion
self.robot_ip_addresses: List[tuple[Lock, str | None]] = [
    (Lock(), robot_ip.get(id, None))
    for id in range(MAX_ROBOT_IDS)
]
```
I suggest writing the code this way because this list contains a list of ip addresses and their associated mutex. This structure will fill out the ip addresses for the robot ID if we have it, otherwise keeps it as None of we discover it layer.